include/deepworks/nn.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <deepworks/placeholder.hpp>
+#include <deepworks/layer_info.hpp>
+#include <deepworks/call.hpp>
+
+namespace deepworks {
+
+template <typename D>
+struct BaseOp {
+    BaseOp(LayerInfo info) : m_info(std::move(info)) { }
+    Placeholder operator()(Placeholder in) {
+        Call call{m_info};
+        call.pass({in});
+        return call.create(static_cast<D*>(this)->output_shape(in.shape()));
+    }
+
+    LayerInfo m_info;
+};
+
+struct Linear : BaseOp<Linear> {
+    Linear(int units, std::string name);
```

Comment:
Is it better to use const std::string& or std::string_view? Same for other ops.

Reply:
The reason of using `std::string` instead of `const string` or `string&&` is having single implementation for both cases. I suppose should be moved to base ctor. I'll check it.

Reply:
Yea, I forgot to add `move` in base `ctor`, thanks!

Fixed